File: backend/test-auth.py
#!/usr/bin/env python3
import requests
import json

def test_auth_endpoints():
    base_url = "http://localhost:5001"
    
    print("🧪 Testing CRM Auth Backend...\n")
    
    try:
        # Test health endpoint
        print("Testing health endpoint...")
        health_response = requests.get(f"{base_url}/health", timeout=5)
        print(f"Health Status: {health_response.status_code}")
        print(f"Health Response: {health_response.text}\n")
        
        # Test registration
        print("Testing registration endpoint...")
        register_data = {
            "name": "Test User",
            "email": "test@example.com",
            "password": "password123"
        }
        
        register_response = requests.post(
            f"{base_url}/auth/register",
            json=register_data,
            timeout=5
        )
        print(f"Register Status: {register_response.status_code}")
        print(f"Register Response: {register_response.text}\n")
        
        # Test login
        print("Testing login endpoint...")
        login_data = {
            "email": "test@example.com",
            "password": "password123"
        }
        
        login_response = requests.post(
            f"{base_url}/auth/login",
            json=login_data,
            timeout=5
        )
        print(f"Login Status: {login_response.status_code}")
        print(f"Login Response: {login_response.text}\n")
        
        print("✅ All tests completed!")
        
    except requests.exceptions.RequestException as e:
        print(f"❌ Test failed: {e}")

if __name__ == "__main__":
    test_auth_endpoints()

File: backend/test-auth.js
#!/usr/bin/env node
const http = require('http');

// Test health endpoint
const testHealth = () => {
  const options = {
    hostname: 'localhost',
    port: 5001,
    path: '/health',
    method: 'GET'
  };

  const req = http.request(options, (res) => {
    console.log(`Health Check Status: ${res.statusCode}`);
    let data = '';
    res.on('data', (chunk) => data += chunk);
    res.on('end', () => {
      console.log('Health Response:', data);
      testRegister();
    });
  });

  req.on('error', (error) => {
    console.error('Health Check Error:', error.message);
  });

  req.end();
};

// Test registration endpoint
const testRegister = () => {
  const postData = JSON.stringify({
    name: 'Test User',
    email: 'test@example.com',
    password: 'password123'
  });

  const options = {
    hostname: 'localhost',
    port: 5001,
    path: '/auth/register',
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      'Content-Length': Buffer.byteLength(postData)
    }
  };

  const req = http.request(options, (res) => {
    console.log(`\nRegister Status: ${res.statusCode}`);
    let data = '';
    res.on('data', (chunk) => data += chunk);
    res.on('end', () => {
      console.log('Register Response:', data);
      testLogin();
    });
  });

  req.on('error', (error) => {
    console.error('Register Error:', error.message);
  });

  req.write(postData);
  req.end();
};

// Test login endpoint
const testLogin = () => {
  const postData = JSON.stringify({
    email: 'test@example.com',
    password: 'password123'
  });

  const options = {
    hostname: 'localhost',
    port: 5001,
    path: '/auth/login',
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      'Content-Length': Buffer.byteLength(postData)
    }
  };

  const req = http.request(options, (res) => {
    console.log(`\nLogin Status: ${res.statusCode}`);
    let data = '';
    res.on('data', (chunk) => data += chunk);
    res.on('end', () => {
      console.log('Login Response:', data);
      console.log('\n✅ All tests completed!');
    });
  });

  req.on('error', (error) => {
    console.error('Login Error:', error.message);
  });

  req.write(postData);
  req.end();
};

console.log('🧪 Testing CRM Auth Backend...\n');
testHealth();